client/src/components/ModalComponents/LocationModal/LocationModal.tsx

```diff
@@ -1,66 +1,74 @@
-// LocationModal.js
-import React from "react";
-import { useDispatch, useSelector } from "react-redux";
-import {
-  setSido,
-  setSigungu,
-  setEupmyeon,
-  addFavoriteLocation,
-  setCurrentLocation,
-} from "../../../redux/locationSlice";
 import * as styles from "./LocationModal.css";
 import locationDatas from "./locationDatas.json";
 import { TiStarOutline, TiStarFullOutline } from "react-icons/ti";
 import { FiX } from "react-icons/fi";
+import { useState } from "react";
+import { toggleLocationModal, useUserInfoStore } from "../../../store/store";
 
 const LocationModal = () => {
-  const dispatch = useDispatch();
-  const { sido, sigungu, eupmyeon, favoriteLocations, currentLocation } = useSelector((state) => state.location);
+  const [state, setState] = useState({
+    sido: "",
+    sigungu: "",
+    eupmyeon: "",
+  });
+  const { favoriteLocations, currentLocation } = useUserInfoStore((state) => ({
+    favoriteLocations: state.favoriteLocations,
+    currentLocation: state.currentLocation,
+  }));
+
+  const { addFavoriteLocation, deleteFavoriteLocation, setCurrentLocation } = useUserInfoStore((state) => ({
+    addFavoriteLocation: state.actions.addFavoriteLocation,
+    deleteFavoriteLocation: state.actions.deleteFavoriteLocation,
+    setCurrentLocation: state.actions.setCurrentLocation,
+  }));
 
   const handleSidoChange = (e) => {
-    dispatch(setSido(e.target.value));
-    dispatch(setSigungu(""));
+    if (e.target.value === "") {
+      return;
+    }
+    setState({ ...state, sido: e.target.value, sigungu: "" });
   };
 
   const handleSigunguChange = (e) => {
-    dispatch(setSigungu(e.target.value));
-  };
+    if (e.target.value === "") {
+      return;
+    }
 
-  const handleEupmyeonChange = (e) => {
-    dispatch(setEupmyeon(e.target.value));
+    setState({ ...state, sigungu: e.target.value });
+    setCurrentLocation(`${state.sido} ${e.target.value}`);
   };
 
-  const handleAddFavorite = () => {
-    dispatch(addFavoriteLocation(currentLocation));
-  };
+  // const handleEupmyeonChange = (e) => {};
 
-  const handleSetCurrentLocation = (location) => {
-    dispatch(setCurrentLocation(location));
+  const handleAddFavorite = () => {
+    if (favoriteLocations.length < 5) {
```

Comment:
아래와 같이 매직넘버는 제거해주시는게 좋습니다~
```suggestion
    if (favoriteLocations.length < FAVORITE_MAX_COUNT) {
```